db/db.go

```diff
@@ -143,23 +136,32 @@ func (db *pgDB) GetUnresolvedBatchKeys(ctx context.Context, limit uint) ([]types
 }
 
 // DeleteUnresolvedBatchKeys deletes the unresolved batch keys from the database
-func (db *pgDB) DeleteUnresolvedBatchKeys(ctx context.Context, bks []types.BatchKey, dbTx sqlx.ExecerContext) error {
+func (db *pgDB) DeleteUnresolvedBatchKeys(ctx context.Context, bks []types.BatchKey) error {
 	const deleteUnresolvedBatchKeysSQL = `
 		DELETE FROM data_node.unresolved_batches
 		WHERE num = $1 AND hash = $2;
 	`
 
+	tx, err := db.pg.BeginTxx(ctx, nil)
+	if err != nil {
+		return err
+	}
+
 	for _, bk := range bks {
-		if _, err := db.execer(dbTx).ExecContext(
+		if _, err = tx.ExecContext(
 			ctx, deleteUnresolvedBatchKeysSQL,
 			bk.Number,
 			bk.Hash.Hex(),
 		); err != nil {
+			if txErr := tx.Rollback(); txErr != nil {
+				return txErr
```

Comment:
Should we include the `err` as well?

E.g.
```suggestion
				return fmt.Errorf("%w: rollback caused by %w", txErr, err)
```

Reply:
Good suggestion, done